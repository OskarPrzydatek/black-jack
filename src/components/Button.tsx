import * as S from '../styles/Buton.style';

interface ButtonProps {
  label: string;
  isDisabled?: boolean;
  dataTestID?: string;
  onClick: () => void;
}

export default function Button({
  label,
  isDisabled,
  dataTestID,
  onClick,
}: ButtonProps) {
  return (
    <S.Button data-testid={dataTestID} disabled={isDisabled} onClick={onClick}>
      {label}
    </S.Button>
  );
}
