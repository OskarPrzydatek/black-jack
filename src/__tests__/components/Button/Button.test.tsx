import renderer from 'react-test-renderer';

import Button from '../../../components/Button';

const mockOnClick = jest.fn();

describe('Button', () => {
  test('component snapshot', () => {
    const tree = renderer
      .create(<Button label="Button" onClick={mockOnClick} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

export {};
