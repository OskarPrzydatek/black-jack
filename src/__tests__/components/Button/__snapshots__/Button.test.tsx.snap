// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component snapshot 1`] = `
<button
  className="sc-bcXHqe etdubz"
  onClick={[MockFunction]}
>
  Button
</button>
`;
